src/v/cloud_storage/remote_segment.cc

```diff
@@ -881,17 +910,91 @@ ss::future<> remote_segment::run_hydrate_bg() {
     _hydration_loop_running = false;
 }
 
-namespace {
+ss::future<> remote_segment::service_chunk_requests() {
+    fragmented_vector<ss::future<ss::file>> chunk_op_results;
+    chunk_op_results.reserve(_chunk_waiters.size());
+
+    fragmented_vector<chunk_request> requests;
+    requests.swap(_chunk_waiters);
+
+    std::ranges::transform(
+      requests,
+      std::back_inserter(chunk_op_results),
+      [this](const auto& request) {
+          vlog(
+            _ctxlog.debug,
+            "Downloading chunk {} with prefetch {}",
+            request.start,
+            request.prefetch);
+          return hydrate_and_materialize_chunk(
+            {request.start, request.prefetch});
+      });
+
+    auto results = co_await ss::when_all(
+      chunk_op_results.begin(), chunk_op_results.end());
+
+    for (size_t i = 0; i < results.size(); ++i) {
+        auto request = std::move(requests[i]);
+        auto current_result = std::move(results[i]);
+
+        ss::file materialized_handle;
+        std::exception_ptr err;
 
+        if (current_result.failed()) {
+            err = current_result.get_exception();
+        } else {
+            materialized_handle = current_result.get();
+        }
+
+        // Materialization may fail due to cache eviction. In this case the
+        // materialized handle is default initialized. Re-queue waiter and try
+        // again later. Continue with other requests in queue.
+        if (!err && !materialized_handle) {
+            vlog(
+              _ctxlog.debug,
+              "Failed to materialize chunk start {}, retrying",
+              request.start);
+            _chunk_waiters.emplace_back(std::move(request));
+            continue;
+        }
+
+        if (err) {
+            request.promise.set_exception(err);
+        } else {
+            request.promise.set_value(materialized_handle);
+        }
+    }
+}
+
+ss::future<ss::file> remote_segment::hydrate_and_materialize_chunk(
+  start_and_prefetch_t start_and_prefetch) {
+    auto [chunk_start, prefetch] = start_and_prefetch;
+    vlog(_ctxlog.debug, "Hydrating chunk {}", chunk_start);
+    co_await hydrate_chunk({_chunks_api->chunk_map(), prefetch, chunk_start});
+    vlog(_ctxlog.debug, "Materializing chunk {}", chunk_start);
+    co_return co_await materialize_chunk(chunk_start);
+}
+
+ss::future<ss::file> remote_segment::download_chunk(
+  chunk_start_offset_t chunk_start, uint16_t prefetch) {
+    auto g = _gate.hold();
```

Comment:
Is the intent here to hold the gate for the duration of the download? If so, we'll need to pass the holder into _chunk_waiters, or `co_return co_await std::move(fut);`

Reply:
It looks like the gate was unnecessary, the remote segment `run_hydrate_bg` already holds a gate which is kept open while the downloads are processed.

Reply:
This gate and co-awaiting to keep it open turned out to be necessary which was causing a new unit test to fail, I have made the changes in the next PR https://github.com/redpanda-data/redpanda/pull/18278/commits/0d99056a86df605b793e4cd31f1170bc36db1460